### ISSUE TITLE
In a net new cluster operator CrashBackLoops until the Red Hat Openshift Pipelines is installed

### ISSUE THREAD

**Issue by acornett21** [redhat-openshift-ecosystem/operator-certification-operator#59]
## Bug Description
If the operator is installed on a cluster that has never had the Red Hat Openshift Pipelines installed on it, the cert operator manager pod will crashback loop until the RH OS Pipelines operator is installed. Once the RH OS Pipelines operator is installed, the cert operator will come up fine and can be installed/uninstalled without the need of the RH OS Pipelines to be present.

## Version and Command Invocation
index image from `latest`

## Steps to Reproduce:
1) Spin up a net new cluster
2) Create a catalog source for the operator
3) Install the operator and try to tail the managers logs or see in `Installed Operators` that the status flips from `Success` -> `Failed` -> `Installing` over and over

## Expected Result
I'd expect that we would be able to install this operator in a net new cluster and not have to install the RH OS Pipelines operator first.

## Actual Result
Controller manager crashback looped and operator install never was successful

## Additional Context
Logs from the controller manager
```
I1118 23:30:21.298373       1 request.go:668] Waited for 1.046372344s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/batch/v1?timeout=32s
2021-11-18T23:30:22.554Z	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "Task.tekton.dev", "error": "no matches for kind \"Task\" in version \"tekton.dev/v1beta1\""}
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/source/source.go:127
2021-11-18T23:30:26.354Z	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "Pipeline.tekton.dev", "error": "no matches for kind \"Pipeline\" in version \"tekton.dev/v1beta1\""}
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/source/source.go:127
2021-11-18T23:30:26.354Z	ERROR	controller-runtime.manager.controller.operatorpipeline	Could not wait for Cache to sync	{"reconciler group": "certification.redhat.com", "reconciler kind": "OperatorPipeline", "error": "failed to wait for operatorpipeline caches to sync: no matches for kind \"Pipeline\" in version \"tekton.dev/v1beta1\""}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:195
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:221
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/manager/internal.go:696
2021-11-18T23:30:26.354Z	ERROR	controller-runtime.manager	error received after stop sequence was engaged	{"error": "leader election lost"}
2021-11-18T23:30:26.354Z	ERROR	setup	problem running manager	{"error": "failed to wait for operatorpipeline caches to sync: no matches for kind \"Pipeline\" in version \"tekton.dev/v1beta1\""}
main.main
	/workspace/main.go:117
runtime.main
	/usr/local/go/src/runtime/proc.go:225
```

**Comment by bcrochet** [comment]
I just tried this on a new cluster, and it worked. I will try again on another cluster (as I had actually manually installed Pipelines operator but then uninstalled)

**Comment by acornett21** [comment]
@bcrochet This is _only_ when the pipelines operator has _never_ been installed in the cluster so the CRD's are not even present for `tekton.Task` / `tekton.Pipeline`. If it's been installed once, we will never fail on install, and on our CR being applied the Pipeliens operator gets installed successfully.

**Comment by bcrochet** [comment]
We need to be able to clean up what we create without using Owns. Might have to be more manual.

**Comment by acmenezes** [comment]
@bcrochet I believe the Owns method sets the controller to listen to events and enqueue work when something happen with the secondary resource configured the SetControllerReference. In this case the tekton types. The cleanup or background deletion can be achieved with SetOwnerReference using the Owns keyword. The side effect is that by using SetOwnerReference only won't trigger the reconciler if something happen with the secondary resource. But deleting the CR will background delete those resources. I guess the solution that seems to be the most reasonable is using the tekton or openshift-pipelines operator as a dependency in OLM. That will guarantee that the CRD is installed, the operator ready and we could still reconcile its secondary resources such as tasks and pipelines. I'm curious about what the team thinks about that.

**Comment by bcrochet** [comment]
I'd rather do the OLM dependency rather than including the tekton CRDs. And if you aren't using OLM, then a prereq is having Tekton already installed.

**Comment by acmenezes** [comment]
Agreed. I think that way too. I'll raise the topic on our next community meeting on Monday.

**Comment by acornett21** [comment]
- Relates #90
